### PR TITLE
Add experiment tracking to RunTracker for PyTorch training loops

### PR DESCRIPTION
**Pull request by rajnisht7**
## Description
Adds an optional `run=` parameter to `visdom.pytorch.VisdomLogger`. So, that whenever `RunTracker` is passed every metric you log also gets saved to that run's local json.

## Motivation and Context
VisdomLogger already plotting training metrics, but it had no way to also keep local reproducibility record of what was logged and when, this connects the two without changing how VisdomLogger works `run=` is not passed.

## How Has This Been Tested?
testing using
```python
from visdom import Visdom
from visdom.pytorch import VisdomLogger
from visdom.tracking import RunTracker

vis = Visdom()

with RunTracker("test_run", params={"lr": 0.01}) as run:
    with VisdomLogger(vis, env="test_run", run=run) as logger:
        for epoch in range(5):
            logger.log("loss", 1.0 / (epoch + 1))

print("status:", run.status)
print("json  :", run.path)
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactor or cleanup (changes to existing code for improved readability or performance)

## Checklist:

- [ ] I adapted the version number under `py/visdom/VERSION` according to [Semantic Versioning](https://semver.org/)
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Summary by Sourcery

Connect VisdomLogger with optional RunTracker recording while keeping plotting and existing ExperimentStore behavior independent and resilient.

New Features:
- Allow VisdomLogger to optionally record plotted metrics in a RunTracker’s local reproducibility data.

Bug Fixes:
- Prevent ExperimentStore failures and tracking warnings from interrupting successful plotting or logging.
- Avoid duplicate metric records when VisdomLogger is combined with an already tracked Visdom client.
- Preserve independent metric window sequences when plotting operates in offline or failed-send modes.

Enhancements:
- Centralize best-effort warning handling so tracking integrations remain non-disruptive under strict warning filters or failing warning hooks.
- Separate plotting, ExperimentStore, and RunTracker error handling so each integration fails independently.

Documentation:
- Document RunTracker integration, lifecycle composition, throttling behavior, failure handling, and duplicate-tracking guidance for VisdomLogger.

Tests:
- Add comprehensive unit coverage for RunTracker integration, ExperimentStore compatibility, throttled and flushed metrics, failure isolation, scalar conversion, warning behavior, and duplicate tracking detection.

## Summary by CodeRabbit

- **New Features**
  - `VisdomLogger` can now record plotted metrics directly to a tracking run.
  - Added guidance for combining run tracking with experiment parameters and avoiding duplicate tracking.

- **Bug Fixes**
  - Tracking warnings no longer interrupt plotting or logging when warnings are configured as errors.
  - Improved resilience when plotting fails or tracking connections are unavailable.
  - Added clearer handling for already-finished runs and double-tracking configurations.

- **Tests**
  - Added comprehensive coverage for run tracking, experiment tracking, throttling, failure handling, warning behavior, and metric logging.

### CODE REVIEW

File: py/tests/unit/pytorch_tracking.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+"""Unit tests for visdom.pytorch.VisdomLogger's optional RunTracker
+integration (the ``run=`` parameter).
+
+Hermetic like tests/unit/tracking*.py: temp dirs for the run's own files,
+and the same _unconnected_visdom() + patched _send() pattern
+tests/unit/plots.py and tests/unit/tracking_graphs.py use to exercise
+Visdom's plotting methods without a real server or network.
+"""
+
```
Comment (tonypzy) [inline py/tests/unit/pytorch_tracking.py:18]: pytest seems not to be imported here

**Review by tonypzy — CHANGES_REQUESTED** [review]
If the caller uses both run.track(viz) and VisdomLogger(..., run=run), the same point will be recorded twice.

**Review by Jayantparashar10 — APPROVED** [review]

**Comment by rajnisht7** [comment]
@sourcery-ai review

**Comment by rajnisht7** [comment]
@sourcery-ai review

**Review by tonypzy — CHANGES_REQUESTED** [review]
DEV was force-overwritten, and it brought in thousands of code changes.

**Review by Jayantparashar10 — APPROVED** [review]

**Review by Manik-Khajuria-5 — CHANGES_REQUESTED** [review]
@rajnisht7 I checked the code and found that the new test file is missing the unit test marker, failed sends can still be recorded as successful, and an invalid run value can cause tracking to crash.
